File: tasks/50_Pow.go
package main

// https://leetcode.com/problems/powx-n
func myPow(x float64, n int) float64 {
	switch {
	case n == 0:
		return 1
	case n < 0:
		return 1 / myPow(x, -n)
	case n%2 == 0:
		half := myPow(x, n/2)
		return half * half
	default:
		return myPow(x, n-1) * x
	}
}

File: tasks/50_Pow.py
# https://leetcode.com/problems/powx-n/

class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n < 0:
            return 1 / self.myPow(x, -n)

        if not n:
            return 1

        half_n, rem = divmod(n, 2)

        rez = self.myPow(x, half_n)

        rez *= rez

        if rem:
            rez *= x

        return rez
